src/lib/components/toast/Toast.component.tsx

```diff
@@ -0,0 +1,180 @@
+import { useToastParameters } from './useToastParameters';
+import styled from 'styled-components';
+
+export type ToastStatus = 'success' | 'error' | 'warning' | 'info';
+
+export type ToastProps = {
+  open: boolean;
+  message: ReactNode;
+  onClose: () => void;
+  status?: ToastStatus;
+  position?: ToastPosition;
+  autoDismiss?: boolean;
+  duration?: number;
+  icon?: React.ReactNode;
+  width?: React.CSSProperties['width'];
+  withProgressBar?: boolean;
+  style?: React.CSSProperties;
+};
+
+export const useGetBackgroundColor = (status: string) => {
+  const theme = useTheme();
+  switch (status) {
+    case 'success':
+      return theme.statusHealthy;
+    case 'error':
+      return theme.statusCritical;
+    case 'warning':
+      return theme.statusWarning;
+    default:
+      return theme.infoPrimary;
+  }
+};
+
+const useGetRgbBackgroundColor = (status: string) => {
+  const theme = useTheme();
+  switch (status) {
+    case 'success':
+      return 'rgba(10, 173, 166, 0.4)';
+    case 'error':
+      return 'rgba(232, 72, 85, 0.4)';
+    case 'warning':
+      return 'rgba(248, 243, 43, 0.4)';
+    default:
+      return theme.infoSecondary;
+  }
+};
+
+const defaultIconName = (status: string) => {
+  switch (status) {
+    case 'success':
+      return 'Check-circle';
+    case 'error':
+      return 'Times-circle';
+    case 'warning':
+      return 'Exclamation-circle';
+    default:
+      return 'Info-circle';
+  }
+};
+
+const DefaultIcon = ({ status }: { status: string }) => {
+  const color = useGetBackgroundColor(status);
+  const iconName = defaultIconName(status);
+
+  return <Icon name={iconName} color={color as IconColor} />;
+};
+
+const DEFAULT_WIDTH = '25rem';
+
+const IconContainer = styled.div<{ bgColor: string }>`
+  align-items: center;
+  align-self: stretch;
+  border-radius: 4px 0px 0px 4px;
+  display: flex;
+  gap: 16px;
+  justify-content: center;
+  position: relative;
+  width: 32px;
+  background-color: ${(props) => props.bgColor};
+`;
+
+const ContentContainer = styled.div`
+  align-items: center;
+  align-self: stretch;
+  display: flex;
+  flex: 1;
+  flex-grow: 1;
+  gap: 8px;
+  padding: 0px 16px;
+  position: relative;
+`;
+
+function Toast({
+  open,
+  message,
+  onClose,
+  position = 'top-right',
+  status = 'info',
+  autoDismiss = true,
+  duration = 5000,
+  icon = <DefaultIcon status={status} />,
+  width = DEFAULT_WIDTH,
+  withProgressBar = false,
+  style,
+}: ToastProps) {
+  const ref = useRef(null);
+  const { params } = useToastParameters({
+    open,
+    duration: autoDismiss ? duration : null,
+    onClose,
+  });
+
+  const positionStyle = positionOutput[position];
+
+  const bgColor = useGetBackgroundColor(status);
+  const rgbBgColor = useGetRgbBackgroundColor(status);
+  const theme = useTheme();
+
+  if (!open) {
+    return null;
+  }
+
+  return (
+    <div
+      ref={ref}
+      role="alert"
+      aria-live="polite"
```

Comment:
👍 does the accessibility spec on toast mention the fact that we need to add a signaler for errors ? something like aria-invalid, but probably not that one as it is reserved for fields iirc

Reply:
https://docs.google.com/document/d/12HoCYmkDvDMATOOJWdryoSwghHFoisA9OQdiPqCqgvY/edit#heading=h.yx1g2yktvxqs not in documentation, you know where is mentionned ?